sbncode/FlashMatch/FlashPredict_module.cc

```diff
@@ -25,64 +25,94 @@ FlashPredict::FlashPredict(fhicl::ParameterSet const& p)
   , fUseOppVolMetric(p.get<bool>("UseOppVolMetric", false))
   , fUseARAPUCAS(p.get<bool>("UseARAPUCAS", false))
-  // , fUseCalo(p.get<bool>("UseCalo", false))
+  , fStoreCheatMCT0(p.get<bool>("StoreCheatMCT0", false))
+    // , fUseCalo(p.get<bool>("UseCalo", false))
   , fInputFilename(p.get<std::string>("InputFileName")) // root file with score metrics
   , fNoAvailableMetrics(p.get<bool>("NoAvailableMetrics", false))
   , fMakeTree(p.get<bool>("MakeTree", false))
   , fChargeToNPhotonsShower(p.get<double>("ChargeToNPhotonsShower", 1.0))  // ~40000/1600
   , fChargeToNPhotonsTrack(p.get<double>("ChargeToNPhotonsTrack", 1.0))  // ~40000/1600
   , fMinHitQ(p.get<double>("MinHitQ", 0.0))
+  , fMinSpacePointQ(p.get<double>("MinSpacePointQ", 0.0))
+  , fMinParticleQ(p.get<double>("MinParticleQ", 0.0))
   , fMinSliceQ(p.get<double>("MinSliceQ", 0.0))
-  , fQScale(p.get<double>("QScale", 1.0))
+  , fMaxFlashes(p.get<unsigned>("MaxFlashes", 1))
   , fMinOpHPE(p.get<double>("MinOpHPE", 0.0))
   , fMinFlashPE(p.get<double>("MinFlashPE", 0.0))
-  , fPEScale(p.get<double>("PEScale", 1.0))
-  , fCryostat(p.get<int>("Cryostat", 0)) //set =0 ot =1 for ICARUS to match reco chain selection
-  , fNBins(p.get<int>("n_bins"))
-  , fDriftDistance(p.get<double>("DriftDistance"))// rounded up for binning
-  , fOpDetNormalizer(p.get<unsigned>("OpDetNormalizer", 1))
+  , fDetector(detectorName(fGeometry->DetectorName()))
+  , fSBND((fDetector == "SBND") ? true : false )
+  , fICARUS((fDetector == "ICARUS") ? true : false )
+  , fPDMapAlgPtr(art::make_tool<opdet::PDMapAlg>(p.get<fhicl::ParameterSet>("PDMapAlg")))
+  , fNTPC(fGeometry->NTPC())
+  , fCryostat(p.get<int>("Cryostat", 0)) //set =0 or =1 for ICARUS to match reco chain selection
+  , fGeoCryo(std::make_unique<geo::CryostatGeo>(fGeometry->Cryostat(fCryostat)))
+  , fWiresX_gl(wiresXGl())
+  , fDriftDistance(driftDistance())
+  , fXBins(p.get<double>("XBins"))
+  , fXBinWidth(fDriftDistance/fXBins)// cm
+  , fRR_TF1_fit(p.get<std::string>("rr_TF1_fit", "pol3"))
+  , fRatio_TF1_fit(p.get<std::string>("ratio_TF1_fit", "pol3"))
+  , fYBiasSlope(p.get<double>("YBiasSlope", 0.))
+  , fZBiasSlope(p.get<double>("ZBiasSlope", 0.))
+  , fOpDetNormalizer((fSBND) ? 4 : 1)
   , fTermThreshold(p.get<double>("ThresholdTerm", 30.))
 {
   produces< std::vector<sbn::SimpleFlashMatch> >();
   produces< art::Assns <recob::PFParticle, sbn::SimpleFlashMatch> >();
-  // fFlashProducer         = p.get<art::InputTag>("FlashProducer");
 
-  fPDMapAlgPtr = art::make_tool<opdet::PDMapAlg>(p.get<fhicl::ParameterSet>("PDMapAlg"));
-  fGeoCryo = std::make_unique<geo::CryostatGeo>(geometry->Cryostat(fCryostat));
-  fNTPC = geometry->NTPC();
-  for (size_t t = 0; t < fNTPC; t++) {
-    const geo::TPCGeo& tpcg = fGeoCryo->TPC(t);
-    fWiresX_gl.push_back(tpcg.LastPlane().GetCenter().X());
+  // TODO: check params are sane:
+  if(fFlashStart > 0. || fFlashEnd < 0.){
+    throw cet::exception("FlashPredict");
```

Comment:
Adding a description explaining why the exception was thrown could save a lot of debugging later down the line.

Reply:
Done.